src/helpers/devserver.ts

```diff
@@ -0,0 +1,87 @@
+import fetch from "cross-fetch";
+import { envKeys, prodEnvKeys } from "./consts";
+import type { FunctionConfig } from "../types";
+
+export const available = async (): Promise<boolean> => {
+  if (isProd()) {
+    return false;
+  }
+
+  try {
+    const url = devserverURL("/dev");
+    const result = await fetch(url.toString());
+    await result.json();
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+// isProd compares any supported standard env variable for "production",
+// returning true on first match.
+export const isProd = (): boolean => {
+  return !!Object.values(prodEnvKeys).find(
+    (e) => process.env[e] === "production"
+  );
+};
+
+// url returns the dev server URL, overriding to use the INNGEST_DEVSERVER_URL
+// env var if provided.
+export const devserverURL = (pathname?: string): URL => {
+  let host = process.env[envKeys.DevServerURL];
```

Comment:
Another `process.env` to remove. Maybe we allow `host` to be passed in instead? 🤔